src/ServicePulse.Host.Tests/tests/js/services/services.monitoring.spec.js

```diff
@@ -28,40 +28,54 @@
 
         var monitoredEndpoints = [];
-        monitoringService.createEndpointsSource(5).subscribe(function (response) {
+        var subscription = monitoringService.createEndpointsSource(5).subscribe(function (response) {
             monitoredEndpoints.push(response);
+
+            if (monitoredEndpoints.length == 2) {
+                expect(monitoredEndpoints[0].name).toEqual("Samples.Metrics.Tracing.Endpoint1");
+                expect(monitoredEndpoints[0].data.timestamps).toEqual([]);
+                expect(monitoredEndpoints[0].data.criticalTime).toEqual([]);
+                expect(monitoredEndpoints[0].data.processingTime).toEqual([]);
+                expect(monitoredEndpoints[1].name).toEqual("Samples.Metrics.Tracing.Endpoint2");
+                expect(monitoredEndpoints[1].data.timestamps).toEqual([]);
+                expect(monitoredEndpoints[1].data.criticalTime).toEqual([]);
+                expect(monitoredEndpoints[1].data.processingTime).toEqual([]);
+
+                subscription.dispose();
+                done();
+            }
         });
-        $httpBackend.flush();
 
-        expect(monitoredEndpoints.length).toEqual(2);
-        expect(monitoredEndpoints[0].name).toEqual("Samples.Metrics.Tracing.Endpoint1");
-        expect(monitoredEndpoints[0].data.timestamps).toEqual([]);
-        expect(monitoredEndpoints[0].data.criticalTime).toEqual([]);
-        expect(monitoredEndpoints[0].data.processingTime).toEqual([]);
-        expect(monitoredEndpoints[1].name).toEqual("Samples.Metrics.Tracing.Endpoint2");
-        expect(monitoredEndpoints[1].data.timestamps).toEqual([]);
-        expect(monitoredEndpoints[1].data.criticalTime).toEqual([]);
-        expect(monitoredEndpoints[1].data.processingTime).toEqual([]);
+        setTimeout(function () {
+            $httpBackend.flush();
+        }, 0);
     });
 
-    it('should push endpoints retrieved from multiple monitoring servers', function () {
-        $httpBackend.whenGET('http://localhost:33633/monitored-endpoints?history=5').respond(monitoredEndpointWithData);
+    it('should push endpoints retrieved from multiple monitoring servers', function (done) {
+        $httpBackend.whenGET('http://localhost:1234/monitored-endpoints?history=5').respond(monitoredEndpointWithData);
         $httpBackend.whenGET('http://localhost:5678/monitored-endpoints?history=5').respond(monitoredEndpointWithData);
 
-        scConfig.monitoring_urls = ['http://localhost:33633/', 'http://localhost:5678/'];
+        scConfig.monitoring_urls = ['http://localhost:1234/', 'http://localhost:5678/'];
 
         var monitoredEndpoints = [];
-        monitoringService.createEndpointsSource(5).subscribe(function (response) {
+        var subscription = monitoringService.createEndpointsSource(5).subscribe(function (response) {
             monitoredEndpoints.push(response);
+
+            if (monitoredEndpoints.length == 4) {
+                expect(monitoredEndpoints.length).toEqual(4);
```

Comment:
This is just to get around the jasmine of no expectations. This test fails if it takes longer then 5 seconds, so the test will still show the failure